### PR TITLE
AI "Go offline" command to replace their brain with a latejoin one

### PR DESCRIPTION
**Pull request by JORJ949**
## About the PR 
Adds a command for AIs to use that ghosts them and replaces their brain with a latejoin one, announcing that they have done so to the crew.

Code probably sucks and I know the warning for doing this also sucks but im PRing this so you can tell me which bits suck

Credit to meaow589 (AKA: Tanuki Singuloose) on the forums for the idea https://forum.ss13.co/showthread.php?tid=23151

## Why's this needed? 
When an AI has to go it can be a pain to replace them or build a latejoin one, so much so that most crews dont bother, this provides an easy way for AI players to leave the game if they need to.

## Changelog 

```changelog
(u)JORJ949
(*) AIs now have a "Go Offline" command to replace themselves with a latejoin AI, this is also announced to the crew. 
```

### CODE REVIEW

File: code/mob/living/silicon/ai.dm [hunk 1]
```diff
@@ -2580,6 +2595,31 @@ proc/get_mobs_trackable_by_AI()
...
 
+// For if an AI needs to disconnect, make their core a latejoin one
+/mob/living/silicon/ai/proc/become_latejoin(var/announce = FALSE)
+	if (deployed_to_eyecam)
+		eyecam.return_mainframe()
+	if (deployed_shell)
+		src.return_to(deployed_shell)
+	src.ghostize()
+	if (src.mind)
+		src.mind.register_death()
```
Comment (TobleroneSwordfish) [inline code/mob/living/silicon/ai.dm:2606]: Ghostize should remove the mind, so this should go above.

File: code/mob/living/silicon/ai.dm [hunk 2]
```diff
@@ -2580,6 +2595,31 @@ proc/get_mobs_trackable_by_AI()
...
+
+	qdel(src.brain)
+	src.brain = new /obj/item/organ/brain/latejoin(src)
+	src.set_color(000000)
+	src.faceEmotion = "ai_blank"
+	src.update_appearance()
+	src.name = "AI"
+	src.UpdateName()
+	if (src.mind)
+		src.mind.assigned_role = "AI"
```
Comment (TobleroneSwordfish) [inline code/mob/living/silicon/ai.dm:2621]: Not sure what this is doing, but again ghostize should have removed the mind by this point

File: code/mob/living/silicon/ai.dm [hunk 3]
```diff
@@ -2580,6 +2595,31 @@ proc/get_mobs_trackable_by_AI()
...
+		eyecam.return_mainframe()
+	if (deployed_shell)
+		src.return_to(deployed_shell)
+	src.ghostize()
+	if (src.mind)
+		src.mind.register_death()
+
+	//Tell the crew the AI is gone
+	if(announce)
+		command_alert("Station AI unit malfunction detected, attempting download of new mind from Central Command database.","Artificial Intelligence Update", alert_origin = ALERT_STATION)
```
Comment (TobleroneSwordfish) [inline code/mob/living/silicon/ai.dm:2610]:
```suggestion
		command_alert("Station AI unit [pick("crash", "kernel panic", "unrecoverable error")] detected, attempting automated download of new personality from Central Command database...","Artificial Intelligence Update", alert_origin = ALERT_STATION)
```
"Malfunction" can mean something very different in the context of AIs...

File: code/mob/living/silicon/ai.dm [hunk 4]
```diff
@@ -2105,6 +2106,20 @@ or don't if it uses a custom topopen overlay
...
 
+/mob/living/silicon/ai/verb/go_offline()
+	set category = "AI Commands"
+	set name = "Go Offline"
+	set desc = "Disconnect your brain such that a new AI can take your place."
+
+	var/mob/message_mob = src.get_message_mob()
+	if (!message_mob.client || isdead(src))
+		return
+	var/confirm = tgui_alert(message_mob, "Become a ghost and permanently replace yourself with a latejoinable AI? (WARNING: YOU CANNOT BE LAWED TO DO THIS AND YOU CANNOT BE REVIVED.)", "Permanently Suicide?", list("Yes", "Cancel"))
```
Comment (TobleroneSwordfish) [inline code/mob/living/silicon/ai.dm:2117]:
```suggestion
	var/confirm = tgui_alert(message_mob, "Become a ghost and allow other players to join into your core? (WARNING: YOU CANNOT BE LAWED OR ORDERED TO DO THIS AND YOU CANNOT BE REVIVED.)", "Permanently Shut Down?", list("Yes", "Cancel"))
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Neat idea, makes sense.

**Comment by Glamurio** [comment]
This is such a scuffed solution, I realize that there's no better way to go about it, but I just find it amusing.

**Review by Tarmunora — APPROVED** [review]

**Comment by FronkyGeniuZ** [comment]
hey it me tanuki thank for PR this appricate alot !

**Pull request merged**